lib/srv/desktop/tdp/proto.go

```diff
@@ -1514,6 +1539,12 @@ func decodeString(r io.Reader, maxLen uint32) (string, error) {
 	return string(s), nil
 }
 
+// writeUint16 writes v to b in big endian order
+func writeUint16(b *bytes.Buffer, v uint16) {
```

Comment:
It seems like you've just reimplemented `binary.Write(b, binary.BigEndian, v)`

In some cases, `binary.Write` should be avoided since it uses reflection which can be slow, but it's probably okay since this channel ID message is once per session.

Alternatively, since this is a pretty simple fixed-size message, you could probably even do without the `bytes.Buffer`:

```
b := make([]byte, 0, 5)
b = append(byte(typeRDPChannelIDs))
b = binary.BigEndian.AppendUint16(c.IOChannelID)
b = binary.BigEndian.AppendUint16(c.UserChannelID)
return b
```

Reply:
This was added for consistency with `writeUint32` and `writeUint64`. Granted we could do without the bytes.Buffer for this message (and others).